test/token_holder/redeem.js

```diff
@@ -0,0 +1,833 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const BN = require('bn.js');
+const EthUtils = require('ethereumjs-util');
+const web3 = require('../test_lib/web3.js');
+const Utils = require('../test_lib/utils.js');
+const { Event } = require('../test_lib/event_decoder');
+const { AccountProvider } = require('../test_lib/utils.js');
+
+const TokenHolder = artifacts.require('TokenHolder');
+const MockRule = artifacts.require('MockRule');
+const TokenRulesMock = artifacts.require('TokenRulesMock');
+const UtilityTokenMock = artifacts.require('UtilityTokenMock');
+
+const ephemeralKeyAddress1 = '0x62502C4DF73935D0D10054b0Fb8cC036534C6fb0';
+const ephemeralPrivateKey1 = '0xa8225c01ceeaf01d7bc7c1b1b929037bd4050967c5730c0b854263121b8399f3';
+const ephemeralPrivateKey2 = '0x634011a05b2f48e2d19aba49a9dbc12766bf7dbd6111ed2abb2621c92e8cfad9';
+
+let token;
+
+
+function getRedeemCallPrefix() {
```

Comment:
This should go to a separate ticket as it will need discussion because we are following above approach(explicit callprefix definition in tests) other tests too. e.g. in executeRule tests:
https://github.com/OpenSTFoundation/openst-contracts/blob/develop/test/token_holder/execute_rule.js#L104

Reason for following it tests:

- Flexibility: CallPrefix can be modified for different kinds of tests(success/failure).
- Readability: After reading the definition of callprefix in tests, devs can understand without going to solidity code.